bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/async/AsyncExecutor.java

```diff
@@ -42,8 +42,9 @@
   public static final int MAX_INFLIGHT_RPCS_DEFAULT = 50;
 
   // This is the maximum accumulated size of uncompleted requests that we allow before throttling.
-  // Default to 32MB.
-  public static final long ASYNC_MUTATOR_MAX_MEMORY_DEFAULT = 16 * 2097152;
+  // Default to 10% of available memory.
+  public static final long ASYNC_MUTATOR_MAX_MEMORY_DEFAULT =
+      (long) (Runtime.getRuntime().maxMemory() * 0.1d);
```

Comment:
"Returns the maximum amount of memory that the Java virtual machine will attempt to use. If there is no inherent limit then the value Long.MAX_VALUE will be returned."

Do you want to protect against the Long.MAX_VALUE case - have some sensible boundary limits.